linked_list/implementation.py

```diff
@@ -7,34 +8,138 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        if elements:
+            for element in elements:
+                self.append(element)
 
     def __str__(self):
-        pass
+        if self.start:
+            current_node = self.start
+            list_string = '['
+            while current_node.next != None:
+                list_string += '{}, '.format(current_node.elem)
+                current_node = current_node.next
+            list_string += '{}]'.format(current_node.elem)
+        else:
+            return '[]'
+        
+        return list_string
 
     def __len__(self):
-        pass
+        count = 0
+        for node in self:
+            count += 1
+        return count
 
     def __iter__(self):
-        pass
+        current_node = self.start
+        while current_node != None:
+            yield current_node
+            current_node = current_node.next
 
     def __getitem__(self, index):
-        pass
+        if self.start == None:
+            raise IndexError
+        if index < 0 or index > self.__len__() - 1:
+            raise IndexError
+        
+        current_node = self.start
+        count = 0
+        while current_node != None:
+            if count == index:
+                return current_node.elem
+            current_node = current_node.next
+            count += 1
+            
 
     def __add__(self, other):
-        pass
+        new_link_list = LinkedList()
+        for node in self:
+            new_link_list.append(node.elem)
+        for node in other:
+            new_link_list.append(node.elem)
+    
+        return new_link_list
 
     def __iadd__(self, other):
-        pass
+        for node in other:
+            self.append(node.elem)
+        return self
 
     def __eq__(self, other):
-        pass
+        if len(self) != len(other):
+            return False
+            
+        for node1, node2 in zip(self, other):
+            if node1 != node2:
+                return False
+        return True
+        
+    def __ne__(self, other):
+        return not self == other
 
     def append(self, elem):
-        pass
+        new_node = Node(elem)
+        if self.start is None:
+            self.start = new_node
+            self.end = new_node
+        else:
+            self.end.next = new_node
+            self.end = new_node
 
     def count(self):
-        pass
+        return self.__len__()
 
     def pop(self, index=None):
```

Comment:
There's a lot of repeated code in here.  You can eliminate pretty much everything in your `if index == None:` by replacing it with
```
if index == None:
    index = len(self) - 1
```
You can then just loop through the list looking for an index match like you do in your final else clause. To handle if you're dealing with start or end you can just check with an if statement to check if the current element is the start(`current == self.start`) or the end(`current.next is None`). At that point you can enter in the case specific handling that your other if/elif statements cover.